systems/subscriber/subscriber-registry/pb/subscriber.proto

```diff
@@ -0,0 +1,190 @@
+syntax = "proto3";
+option go_package = "pb/gen";
+
+import "google/protobuf/timestamp.proto";
+package ukama.subscriber.subscriber_registry.v1;
+import "validator.proto";
+
+/*
+Subscriber-registry sub-system is responsible of:
+    - Creating subscriber
+    - Update subscriber
+    - Get subscriber by network
+    - Get single subscriber with Sims associated to them
+*/
+// Defines the service for subscriber registry operations
+service SubscriberRegistryService {
+// Get method to retrieve a subscriber by subscriber ID
+rpc Get (GetSubscriberRequest) returns (GetSubscriberResponse) {}
+// Add method to add a new subscriber
+rpc Add (AddSubscriberRequest) returns (AddSubscriberResponse){}
+// Update method to update an existing subscriber
+rpc Update(UpdateSubscriberRequest) returns (UpdateSubscriberResponse){}
+// Delete method to delete a subscriber by subscriber ID
+rpc Delete (DeleteSubscriberRequest) returns (DeleteSubscriberResponse){}
+// GetByNetwork method to retrieve subscribers by network ID
+rpc GetByNetwork(GetByNetworkRequest) returns(GetByNetworkResponse){}
+// ListSubscribers method to retrieve a list of all subscribers
+rpc ListSubscribers(ListSubscribersRequest) returns(ListSubscribersResponse){}
+}
+
+// ListSubscribersRequest message is used to list all subscribers
+message ListSubscribersRequest{
+}
+
+// ListSubscribersResponse message is used to list all subscribers
+message ListSubscribersResponse{
+// Repeated field of Subscriber message
+repeated Subscriber subscribers = 1;
+}
+
+// DeleteSubscriberRequest message is used to delete a subscriber
+message DeleteSubscriberRequest{
+// subscriberID field is used to specify the subscriber id for deletion
+string subscriberID = 1 [(validator.field) = {uuid_ver: 0, string_not_empty: true}, json_name = "subscriber_id"];
+}
+
+// GetByNetworkRequest message is used to get all subscribers by network id
+message GetByNetworkRequest{
+// networkID field is used to specify the network id for getting subscribers
+string networkID = 1 [(validator.field) = {uuid_ver: 0, string_not_empty: true}, json_name = "network_id"];
+
+}
+
+// GetByNetworkResponse message is used to get all subscribers by network id
+message GetByNetworkResponse{
+// Repeated field of Subscriber message
+repeated Subscriber subscribers = 1;
+}
+
+// DeleteSubscriberResponse message is used to delete a subscriber
+message DeleteSubscriberResponse{
+// subscriberID field is used to specify the subscriber id for deletion
+string subscriberID = 1;
```

Comment:
Why do we return the ID of the deleted subscriber, especially if that ID is no more valid (because the subscriber is deleted) ? And also, we already have this information as this is the ID that we pass to the delete request.